### ISSUE TITLE
dotCMS 20+ Error on changing default language

### ISSUE THREAD

**Issue by miguel876** [dotCMS/core#20112]
In the versions 20+ and 21+, there's a bug when changing the default language following the official documentation.

https://dotcms.com/docs/latest/system-language-properties

After changing the default language, some components don't seem to work, for example, in the templates menu, the container dropdown is empty. There are also error messages when acessing the Users, Sites and Content Types menus.

The change made was to add the following lines to the **dotmarketing-config.properties** file and replaced the default ones:

LANGUAGE AND CODING INFO
CHARSET = text/html;charset=UTF-8
DEFAULT_LANGUAGE_COUNTRY = Portugal
DEFAULT_LANGUAGE = pt_PT
DEFAULT_LANGUAGE_CODE = pt
DEFAULT_LANGUAGE_STR = Portugues
DEFAULT_LANGUAGE_COUNTRY_CODE = PT

**How to reproduce the incident:**
1. Change the default language following the documentation.
2. Access dotCMS and the pages Users and Sites. Try to add a container to a template.

**Screenshots of the incident:**
![image](https://user-images.githubusercontent.com/25934089/111651345-0d543f00-87fe-11eb-9ce7-b131174227ff.png)
![image](https://user-images.githubusercontent.com/25934089/111651372-11805c80-87fe-11eb-883d-45899afd755d.png)
![image](https://user-images.githubusercontent.com/25934089/111651383-134a2000-87fe-11eb-9b87-13b1a3bf81d6.png)

Thank You

**Comment by wezell** [comment]
@miguel876 can you try with the empty starter?  Any files that have been created in dotcms before the language has been switched (or are included in the starter) need to be recreated I believe

**Comment by miguel876** [comment]
Thanks for the reply @wezell , 

I tried that approach and it did not work.
I unzipped the dotCMS 21.01 folder, did the basic configurations and added the new language configurations.

![image](https://user-images.githubusercontent.com/25934089/111688868-3b984580-8823-11eb-8cef-180896a537da.png)

 After that I started dotCMS but got the following errors:

![image](https://user-images.githubusercontent.com/25934089/111688245-0d1a6a80-8823-11eb-89da-d1677be66b92.png)

**Comment by erickgonzalez** [comment]
Related to: https://github.com/dotCMS/core/issues/18766

**Comment by miguel876** [comment]
Thank you for the reply @wezell,

I started from scratch, downloaded dotCMS version 21.04 (latest version at the moment), created a clean database (PostgreSQL) and resumed the configuration. After configuring everything i started dotCMS, logged in, created a new language (PT in my case), and shut down dotCMS. Then I updated all the **language_id** fields on my database to the new language id (PT). I started dotCMS again and it still gives me the same errors that I previously commented and I've also crossed some errors while attempting to reindex.

WARN mapper.JsonMappingExceptionMapper: No version info. Call setWorking first 8a7d5e23-da1e-420a-b4f0-471e7da8ea2d (through reference chain: com.dotcms.rest.ResponseEntityView["entity"]->com.dotmarketing.beans.Host["live"])

I also got some errors on the browser console log.

![image](https://user-images.githubusercontent.com/25934089/118489216-7da31200-b714-11eb-882c-d8a0a76b5e3d.png)

**Comment by erickgonzalez** [comment]
Hi @miguel876 ,

I'm working on this one at this moment, we'll have a fix for the next release (21.06). As soon I gather more info, I'll share with you if there is any workaround at the time to get through this issue you are getting.

**Comment by erickgonzalez** [comment]
Hi @miguel876,

After doing some research I was able to fix the issue that you are hitting I will create a PR with the fix and will be shipped in the next release (21.06). Meanwhile, can you try the following steps:

1. Startup dotcms using our default language value (en_US).
2. Create the new language that you want to be your new default one (PT in your case).
3. Save the id of the created language in a note somewhere, we need it for the next step.
4. Stop dotcms.
5. Run the following queries against the DB, change **{langId}** for the id of the language (step 3).
```
update contentlet set language_id = {langId} where language_id = '1' and identifier not in (select identifier from contentlet where language_id = {langId});
update contentlet_version_info set lang = {langId} where lang = 1 and identifier not in (select identifier from contentlet_version_info where lang = {langId});
update workflow_task set language_id = {langId} where language_id = 1 and webasset not in (select webasset from workflow_task where language_id = {langId});
```
6. Change **dotmarketing-config.properties** to the new default language:
```
DEFAULT_LANGUAGE_COUNTRY = Portugal
DEFAULT_LANGUAGE = pt_PT
DEFAULT_LANGUAGE_CODE = pt
DEFAULT_LANGUAGE_STR = Portugues
DEFAULT_LANGUAGE_COUNTRY_CODE = PT
```
- Startup again the dotcms instance.
- Flush the cache and run a full reindex.

Hopefully, that will do the trick.

**Comment by miguel876** [comment]
Thanks @erickgonzalez for the reply!

I was able to change the default language without any errors after doing the steps you commented. The problem was in my db querys I believe.

**Comment by bryanboza** [comment]
Fixed, tested on release-21.06 following the steps from the @erickgonzalez comment

**Comment by jcastro-dotcms** [comment]
Excluding from dotCMS 5.3.8.6LTS patch because no new customer deployments will be using 5.3.8.6.
